apps/ui/pages/reservations/[id]/cancel.tsx

```diff
@@ -54,15 +61,17 @@ export async function getServerSideProps(ctx: GetServerSidePropsContext) {
     );
 
     // TODO check for null and return notFound
-    return {
-      props: {
-        ...commonProps,
-        ...(await serverSideTranslations(locale ?? "fi")),
-        key: `${pk}-cancel-{locale}`,
-        reservation: reservation ?? null,
-        reasons,
-      },
-    };
+    if (reservation != null && reservation.user?.pk === user?.pk) {
+      return {
+        props: {
+          ...commonProps,
+          ...(await serverSideTranslations(locale ?? "fi")),
+          key: `${pk}-cancel-{locale}`,
+          reservation: reservation ?? null,
```

Comment:
can remove `??...`

Reply:
yeah,`?? null` seems to be a bit redundant, but `?? "fi"` seems to be necessary as a fallback